### ISSUE TITLE
`get_upgrade_versions` endpoint

### ISSUE THREAD

**Issue by ikalnitsky** [sc68cal/Kostyor#72]
I've stumbled upon the following endpoint:

```python
@app.route('/upgrade-versions/<cluster_id>')
def get_upgrade_versions(cluster_id):
    cluster = db_api.get_cluster(cluster_id)
    if not cluster:
        resp = generate_response(404, 'Cluster %s not found' % cluster_id)
        return resp
    cluster_version_index = constants.OPENSTACK_VERSIONS.index(
        cluster['version'])
    upgrade_versions = (
        constants.OPENSTACK_VERSIONS[cluster_version_index + 1:])
    return jsonify(upgrade_versions)
```

I was considering to move it to Flask-RESTful and came up with some questions. Here they are:

* Which resource should return this sort of information: `/clusters` or `/upgrades`?
* Do we need a separate sub-resource for this job (e.g. `/clusters/<cluster_id/upgrade-version`) or it would be enough to return this information as a part of Cluster response? Example:

 ```http
  GET /clusters/:cluster_id HTTP/1.1

  {
      "id": "7da361cd-48d9-4b2f-bdc6-80bd473ff12a",
      "name": "devstack",
      "status": "READY FOR UPGRADE",
      "version": "mitaka",
      "upgrade_versions": [

          "newton",
          "ocata"
      ]
  }
  ```

* Does it make sense to return multiple possible upgrades? As far as I understand, OpenStack supports upgrades (or at least suppose to 😄 ) only for neighbor releases (i.e. mitaka -> newton -> ocata).

Any feedback, any thoughts are really appreciated and valuable.

**Comment by sc68cal** [comment]
Let's defer this, as I said about the other REST API work. https://github.com/sc68cal/Kostyor/issues/73#issuecomment-254848251

**Comment by ikalnitsky** [comment]
@sc68cal We definitely should postpone this till 2017, though I'd like to hear opinions and gather some feedback. We can have some discussion here in order to figure out what do we want in the end. :)

**Comment by sc68cal** [comment]
@kalnytskyi agreed. When we hit Q1 let's do that. I'm taking a leaf out of your book and doing the same for the openstack-ansible discovery driver - open an issue, and discuss before coding :)
